test/src/test/java/org/corfudb/infrastructure/logreplication/LogReplicationFSMTest.java

```diff
@@ -155,6 +175,128 @@ public void testLogReplicationFSMTransitions() throws Exception {
         transition(LogReplicationEventType.REPLICATION_STOP, LogReplicationStateType.INITIALIZED, true);
     }
 
+    /**
+     * Verify the lastSyncType flag is being initialized properly and SnapshotSyncInfo
+     * reflects accurate sync type for snapshot sync.
+     *
+     */
+    @Test
+    public void testSyncStatusUpdatesForSnapshotOnInit() throws Exception {
+        final int updateToStatusTableFromOnEntry = 1;
+        initLogReplicationFSM(ReaderImplementation.EMPTY);
+
+        final Table<ReplicationStatusKey, ReplicationStatusVal, Message> statusTable =
+                this.corfuStore.getTable(NAMESPACE, REPLICATION_STATUS_TABLE);
+
+        CountDownLatch statusTableLatch = new CountDownLatch(updateToStatusTableFromOnEntry);
+        TestStatusTableStreamListener streamListener = new TestStatusTableStreamListener(statusTableLatch);
+        corfuStore.subscribeListener(streamListener, NAMESPACE, LR_STATUS_STREAM_TAG);
+
+        // Initial state: Initialized
+        LogReplicationState initState = fsm.getState();
+        assertThat(initState.getType()).isEqualTo(LogReplicationStateType.INITIALIZED);
+
+        transitionAvailable.acquire();
+
+        ReplicationStatusVal currentReplicationVal = statusTable.entryStream().findAny().get().getPayload();
```

Comment:
the record must be fetched by constructing the key and using getRecord() so that we validate the right one.  The test currently has only 1 record so will not fail but can change in future.